### PR TITLE
Add missing isos to `global`

### LINKED ISSUE #182: Difference in Corporate Economy In the results vs the ALD
There is a slight difference between the corporate economy as calculated by r2dii.analysis::target_market_share and the sum of production from the ALD. 

I believe that this is negligible and potentially due to rounding somewhere in the code. 

I raise it non the less in case it is an easy fix or if any users question this. 

Attached proves that r2dii.analysis::target_market_share calculates a consistent corporate economy across portfolios. 
I have shown the percentage difference between the corporate economy in both data sets (results and ALD) - I have taken examples of from different sectors across different years. 
The largest percentage difference comes from ICE in the auto sector = **1.33% difference**

Thanks 
[ALD_vs_targets_corp_econ.xlsx](https://github.com/2DegreesInvesting/r2dii.analysis/files/5582912/ALD_vs_targets_corp_econ.xlsx)

### PR DESCRIPTION
**Pull request by jdhoffa**
I was determined that some `isos` that are present in the release version of the `ald` data, are not accounted for in the `global` region of our scenarios in `r2dii.data::region_isos`. 

`global` should account for every country. I'm not sure the most reproducible way to do this, but what I've done here is: 
* Manually determine all unique `isos` in the release `ald` data
* Save this to `data-raw/all_ald_isos.csv` (happy to rename this)
* Ensure that these `isos` are appended to the end of `region_isos` when sourcing that data

@maurolepore I'm not sure if this is a great solution, so please feel free to comment. 

Closes #182

### CODE REVIEW

File: data-raw/region_isos.R [hunk 1]
```diff
@@ -91,6 +91,11 @@ exclude_values <- function(data, values, .region) {
     mutate(region = .region)
 }
 
+# Some isos are missing from the global regions definition. Here we read in the
+# actual isos from a real ALD file, and buffer the potentially missing isos.
+ald_path <- file.path("data-raw", paste0("ald_all_isos", ".csv"))
```
Comment (maurolepore) [inline data-raw/region_isos.R:96]: Is `ald_path` a good name to hold a path to ald_all_isos.csv?
If finding a good name is hard, maybe inline into `read_regions()`?

File: data-raw/region_isos.R [hunk 2]
```diff
@@ -155,7 +160,9 @@ region_isos_weo_2019 <- bound1 %>%
     non_oecd,
     non_opec
   ) %>%
-  prepare_isos()
+  prepare_isos() %>%
+  bind_rows(mutate(ald_isos, source = "weo_2019")) %>%
```
Comment (maurolepore) [inline data-raw/region_isos.R:164]: Nit: Maybe `mutate` in line 97-98, where we are still concerned with details about `ald_isos`. Here I would prefer to read code at a higher level of abstraction. Something that would make sense to me would be ...

```
...
  prepare_isos() %>%
  bind_rows(isos_including_missing_isos) %>%
...
```

**Review by maurolepore — APPROVED** [review]
LGTM. I made a couple of Nit comments you may disregard. 
Thanks!

File: data-raw/region_isos.R (hunk 2 above)
Comment (jdhoffa) [reply data-raw/region_isos.R:164]: ya i was thinking that too. the issue is that it's easiest to do two separate mutates, one for `weo_2019` and one for `etp_2017`

Comment (jdhoffa) [reply data-raw/region_isos.R]: see 836035c

**Comment by maurolepore** [comment]
Anything here deserves a place in NEWS.md?

**Comment by jdhoffa** [comment]
Yup good shout! news updated.

**Pull request merged**